app/src/main/java/net/gsantner/markor/ui/hleditor/HighlightingEditor.java

```diff
@@ -10,116 +10,116 @@
 package net.gsantner.markor.ui.hleditor;
 
 import android.content.Context;
+import android.graphics.Rect;
 import android.os.Build;
-import android.os.Handler;
 import android.support.annotation.RequiresApi;
 import android.support.v7.widget.AppCompatEditText;
 import android.text.Editable;
 import android.text.InputFilter;
+import android.text.Layout;
 import android.text.TextWatcher;
 import android.util.AttributeSet;
 import android.view.KeyEvent;
 import android.view.View;
+import android.view.ViewTreeObserver;
 import android.view.accessibility.AccessibilityEvent;
 
 import net.gsantner.markor.activity.MainActivity;
-import net.gsantner.markor.model.Document;
 import net.gsantner.markor.util.AppSettings;
 import net.gsantner.opoc.util.Callback;
 import net.gsantner.opoc.util.StringUtils;
 
-import java.io.File;
 import java.util.HashSet;
 import java.util.Set;
 
 @SuppressWarnings("UnusedReturnValue")
-public class HighlightingEditor extends AppCompatEditText {
+public class HighlightingEditor extends AppCompatEditText implements ViewTreeObserver.OnScrollChangedListener, TextWatcher {
+
+    final static float SIGNIFICANT_SHIFT_FRACTION = 0.2f;
```

Comment:
is this scroll screen area treshhold multiplier?

Reply:
Not quite. Screen area threshold is set to +-1 right now. This is a threshold s.t. the user has to scroll > 0.2x the currently highlighted region to trigger a reapplication. Prevents reapplication of highlight on small jumps and scrolls as you type.

Reply:
Note: I have updated this now. Still experimenting to find good thresholds. Also need to apply a small file optimization where we don't do the dynamic highlighting for files under, say, 25k characters.

Reply:
How do you feel about +-2  x screensize?